### PR TITLE
Mention affiliated packages in CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by abigailStev**
Added small paragraph about contributing to affiliated packages and a link to the affiliated packages page. Also added a bullet point under Tips about there being no naming convention for working branches. Fixes https://github.com/astropy/astropy/issues/6866.

### CODE REVIEW

**Comment by abigailStev** [comment]
Ping @bsipocz

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -80,6 +88,9 @@ Other Tips
 
       $ git commit -m "Fixed typo [docs only]"
 
+- There is no naming standard for branches, but don't name your working
```
Comment (bsipocz) [inline CONTRIBUTING.md]: This is not needed here I feel, and it's already in the development workflow linked above.

**Comment by abigailStev** [comment]
Done, will merge when tests pass!

**Comment by bsipocz** [comment]
Thanks @abigailStev!

**Pull request merged**